src/Crawler.php

```diff
@@ -314,7 +319,11 @@ public function setBrowsershot(Browsershot $browsershot)
     public function setUserAgent(string $userAgent): Crawler
     {
         $clientOptions = $this->client->getConfig();
-        $clientOptions['headers']['User-Agent'] = strtolower($userAgent);
+
+        $headers = array_change_key_case($clientOptions['headers']);
```

Comment:
Why do we need this, if we're already doing lowercase checks in `getUserAgent`?

Reply:
This is necessary because we want to be sure that we are adding a new header or updating one that already exists. Otherwise there is a risk that we will duplicate it and which one should be returned by the `getUserAgent` method then? This is not too obvious in my opinion. Guzzle has similar implementation https://github.com/guzzle/guzzle/blob/master/src/Client.php#L196

Reply:
Doesn't that mean that this check is redundant? However you construct a client:

```php
$this->client = new Client($clientOptions);
```

Guzzle will make internally sure there will never be two `user-agent` headers because of the check you linked (https://github.com/guzzle/guzzle/blob/master/src/Client.php#L198)

Reply:
You may be right, though this should be unit tested. I'll take care of it.

Reply:
Awesome, thanks!

Reply:
@brendt, I'm afraid that without changing array keys case, the following test:
```php
public function it_should_ignore_user_agents_header_case()
{
    $clientConfig = ['headers' => ['user-agent' => 'foo']];
    $newUserAgent = 'bar';

    $crawler = Crawler::create($clientConfig)->setUserAgent($newUserAgent);
    $actualUserAgent = $crawler->getUserAgent();

    $this->assertSame($newUserAgent, $actualUserAgent);
}
```
fails with message:
```
Failed asserting that two strings are identical.
Expected :'bar'
Actual   :'foo'
```

Reply:
No problem, we'll merge it like this